packages/hint-no-broken-links/src/hint.ts

```diff
@@ -169,15 +146,56 @@ export default class NoBrokenLinksHint implements IHint {
             fetchedUrls.push({ statusCode: fetchEnd.response.statusCode, url: fetchEnd.resource });
         };
 
-        context.on('element::img', validateElementSrcs);
-        context.on('element::a', validateElementSrcs);
-        context.on('element::audio', validateElementSrcs);
-        context.on('element::video', validateElementSrcs);
-        context.on('element::link', validateElementSrcs);
-        context.on('element::script', validateElementSrcs);
-        context.on('element::source', validateElementSrcs);
-        context.on('element::track', validateElementSrcs);
-        context.on('element::object', validateElementSrcs);
+        const validateCollectedUrls = async (event: TraverseEnd) => {
+            const { resource } = event;
+            const pageDOM: IAsyncHTMLDocument = context.pageDOM as IAsyncHTMLDocument;
+            const baseTags: Array<IAsyncHTMLElement> = await pageDOM.querySelectorAll('base');
+            const hrefAttribute = (baseTags.length === 0) ? null : baseTags[0].getAttribute('href');
+
+            const reports: Array<Promise<void>> = collectedElementsWithUrls.reduce<Promise<void>[]>((accumulatedReports, [element, urls]) => {
+                return [...accumulatedReports, ...urls.map(async (url) => {
+                    const isRelativeUrl = (!url.startsWith('/') && URL.parse(url).hostname === null);
+                    const fullUrl = (isRelativeUrl && hrefAttribute !== null) ?
+                        URL.resolve(resource, hrefAttribute + url) :
+                        URL.resolve(resource, url);
+
+                    const fetched = getFetchedUrl(fullUrl);
+
+                    if (fetched) {
+                        const statusIndex = brokenStatusCodes.indexOf(fetched.statusCode);
+
+                        if (statusIndex > -1) {
+                            return context.report(fullUrl, null, `Broken link found (${brokenStatusCodes[statusIndex]} response).`);
+                        }
+                    } else {
+                        // An element which was not present in the fetch end results
+                        return await requester
```

Comment:
What do you think about removing the `async/await` from this function. I know it's in the original code but I think it can be safely removed.

Reply:
Yes, you're right. There seems to be no reason for awaiting that promise.